### ISSUE TITLE
[Accessibility]: Incorrect focus and announcement on screen reader clicking on an occurrence

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#36563]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33509

### Problem description

On performing a screen reader click on an occurrence in the blob view, the focus shifts to the entire page (instead of the popover) and the page is announced.

(Putting this under "Go to Definition" but it also applies to "Find References" and related user journeys.)

### Expected behavior

The focus should shift to the code intel popover and the change in focus should be announced in a clear way.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by umpox** [comment]
This also falls under general blob view keyboard navigation.

Closing this in favour of:
- https://github.com/sourcegraph/sourcegraph/issues/42420
- https://github.com/sourcegraph/sourcegraph/issues/42421
